### PR TITLE
PROG2-28: Add WatchlistEntity class

### PR DESCRIPTION
**Pull request by burakkongo**
Add WatchlistEntity class and alter pom.xml and module-info.java to use h2Database and j256 ormlite for the in-memory database usage.

### CODE REVIEW

File: src/main/java/at/ac/fhcampuswien/fhmdb/models/WatchlistEntity.java [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    public WatchlistEntity(Movie movie) {
+        this.apiId = movie.getId();
+        this.imgUrl = movie.getImgUrl();
+        this.title = movie.getTitle();
+        this.description = movie.getDescription();
+        this.genres = genreListToString(movie.getGenres());
+        this.releaseYear = movie.getReleaseYear();
+        this.directors = listToString(movie.getDirectors());
+        this.writers = listToString(movie.getWriters());
+        this.mainCast = listToString(movie.getMainCast());
```
Comment (sorinlzr) [inline src/main/java/at/ac/fhcampuswien/fhmdb/models/WatchlistEntity.java:64]: According to the Class diagram, these fields are not needed, or am I missing something?

**Review by sorinlzr — CHANGES_REQUESTED** [review]

File: src/main/java/at/ac/fhcampuswien/fhmdb/models/WatchlistEntity.java (hunk 1 above)
Comment (burakkongo) [reply src/main/java/at/ac/fhcampuswien/fhmdb/models/WatchlistEntity.java:64]: I am mapping the data to the Database fields. As Ive understood it you have to declare these database fields and then map the data to them.

fe.:
@DatabaseField
    private String directors;

    @DatabaseField
    private String writers;

    @DatabaseField
    private String mainCast;

    @DatabaseField
    private double rating;

Comment (burakkongo) [reply src/main/java/at/ac/fhcampuswien/fhmdb/models/WatchlistEntity.java:64]: and we cannot save a list in the database. We save it as a String with the comma delimiter

Comment (sorinlzr) [reply src/main/java/at/ac/fhcampuswien/fhmdb/models/WatchlistEntity.java:64]: I understand, but in the Angabe it says "Da für das Speichern von Listen (zB directors, writers, mainCast) weitere Tabellen nötig wären, werden diese ausgenommen.", that's why I asked

Comment (burakkongo) [reply src/main/java/at/ac/fhcampuswien/fhmdb/models/WatchlistEntity.java:64]: I dient want to have the same shitty situation as last time with the UI thing where we lost points so I have implemented it ^^.

Comment (sorinlzr) [reply src/main/java/at/ac/fhcampuswien/fhmdb/models/WatchlistEntity.java:64]: Fair enough 😁

**Review by sorinlzr — APPROVED** [review]
~let's gooo~ lgtm

**Pull request merged**